ee/app/livechat-enterprise/server/api/units.ts

```diff
@@ -0,0 +1,92 @@
+import { API } from '../../../../../app/api/server';
+import { findUnits, findUnitById, findUnitMonitors } from './lib/units';
+import { LivechatEnterprise } from '../lib/LivechatEnterprise';
+import { IOmnichannelBusinessUnit } from '../../../../../definition/IOmnichannelBusinessUnit';
+
+API.v1.addRoute('livechat/units.list', { authRequired: true }, {
+	async get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort } = this.parseJsonQuery();
+		const { text } = this.queryParams;
+
+		const response = await findUnits({
+			userId: this.userId,
+			text,
+			pagination: {
+				offset,
+				count,
+				sort,
+			},
+		});
+
+		return API.v1.success(response);
+	},
+});
+
+API.v1.addRoute('livechat/units.getOne', { authRequired: true }, {
+	async get() {
+		const { id } = this.urlParams;
+		const { unit } = await findUnitById({
+			userId: this.userId,
+			unitId: id,
+		}) as { unit: IOmnichannelBusinessUnit };
+
+		return API.v1.success(unit);
+	},
+});
+
+API.v1.addRoute('livechat/unitMonitors.list', { authRequired: true }, {
+	async get() {
+		const { unitId } = this.queryParams;
+
+		return API.v1.success(await findUnitMonitors({
+			userId: this.userId,
+			unitId,
+		}));
+	},
+});
+
+API.v1.addRoute('livechat/units', { authRequired: true, permissionsRequired: ['manage-livechat-units'] }, {
+	async get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort } = this.parseJsonQuery();
+		const { text } = this.queryParams;
+
+		return API.v1.success(Promise.await(findUnits({
+			userId: this.userId,
+			text,
+			pagination: {
+				offset,
+				count,
+				sort,
+			},
+		})));
+	},
+	async post() {
+		const { unitData, unitMonitors, unitDepartments } = this.bodyParams?.();
+		return LivechatEnterprise.saveUnit(null, unitData, unitMonitors, unitDepartments);
+	},
+});
+
+API.v1.addRoute('livechat/units/:id', { authRequired: true, permissionsRequired: ['manage-livechat-units'] }, {
+	async get() {
```

Comment:
Same goes here. I think we already have an endpoint which fetches this data - `livechat/units.getOne` (GET)

Reply:
Yup, the original idea was to mark the others as deprecated in favor of these ones (which are a bit more restful than `.getOne`, etc) but I had some issues that I'm still solving with the deprecationWarning helper.

Reply:
Ahh.. okay makes sense. It's fine then.